testsuite/testcases/src/three_region_simulation_test.rs

```diff
@@ -99,6 +108,67 @@ fn create_bandwidth_limit() -> SwarmNetworkBandwidth {
     }
 }
 
+fn add_execution_delay(swarm: &mut dyn Swarm, config: &ExecutionDelayConfig) -> anyhow::Result<()> {
+    let runtime = Runtime::new().unwrap();
+    let validators = swarm.get_validator_clients_with_names();
+
+    runtime.block_on(async {
+        let mut rng = rand::thread_rng();
+        for (name, validator) in validators {
+            let sleep_fraction = if rng.gen_bool(config.inject_delay_node_fraction) {
+                rng.gen_range(1_u32, config.inject_delay_max_transaction_percentage)
+            } else {
+                0
+            };
+            let name = name.clone();
+            info!(
+                "Validator {} adding {}% of transactions with 1ms execution delay",
+                name, sleep_fraction
+            );
+            validator
```

Comment:
nit: seems like you could move this into a function to avoid the duplication

Reply:
you mean the set_failpoint? well arguments are different, and map_err is for context so different, so not clear worth extracting into function